File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/SystemShareHandler.java
package closer.vlllage.com.closer.handler.helpers;

import android.content.Intent;
import android.graphics.Bitmap;
import android.net.Uri;
import android.support.v4.content.FileProvider;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;

import closer.vlllage.com.closer.R;
import closer.vlllage.com.closer.handler.media.MediaHandler;
import closer.vlllage.com.closer.pool.PoolMember;

public class SystemShareHandler extends PoolMember {

    public void share(Bitmap bitmap) {
        if (bitmap == null) {
            $(DefaultAlerts.class).thatDidntWork();
            return;
        }

        File file;
        try {
            file = $(MediaHandler.class).createTemporaryFile("closer-photo", ".jpg");

            if (file == null) {
                $(DefaultAlerts.class).thatDidntWork();
                return;
            }

            FileOutputStream stream = new FileOutputStream(file);
            bitmap.compress(Bitmap.CompressFormat.JPEG, 91, stream);
            stream.close();
        } catch (IOException e) {
            e.printStackTrace();
            $(DefaultAlerts.class).thatDidntWork();
            return;
        }

        Uri contentUri = FileProvider.getUriForFile($(ActivityHandler.class).getActivity(), MediaHandler.AUTHORITY, file);

        if (contentUri == null) {
            $(DefaultAlerts.class).thatDidntWork();
            return;
        }

        Intent shareIntent = new Intent();
        shareIntent.setAction(Intent.ACTION_SEND);
        shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
        shareIntent.setDataAndType(contentUri, $(ApplicationHandler.class).getApp().getContentResolver().getType(contentUri));
        shareIntent.putExtra(Intent.EXTRA_STREAM, contentUri);
        shareIntent.setType("image/jpg");
        $(ActivityHandler.class).getActivity().startActivity(Intent.createChooser(shareIntent, $(ResourcesHandler.class).getResources().getString(R.string.share_to)));
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/SystemShareHandler.kt
package closer.vlllage.com.closer.handler.helpers

import android.content.Intent
import android.graphics.Bitmap
import androidx.core.content.FileProvider
import closer.vlllage.com.closer.R
import closer.vlllage.com.closer.handler.media.MediaHandler
import com.queatz.on.On
import java.io.File
import java.io.FileOutputStream
import java.io.IOException

class SystemShareHandler constructor(private val on: On) {

    fun share(bitmap: Bitmap?) {
        if (bitmap == null) {
            on<DefaultAlerts>().thatDidntWork()
            return
        }

        val file: File?
        try {
            file = on<MediaHandler>().createTemporaryFile("closer-photo", ".jpg")

            if (file == null) {
                on<DefaultAlerts>().thatDidntWork()
                return
            }

            val stream = FileOutputStream(file)
            bitmap.compress(Bitmap.CompressFormat.JPEG, 91, stream)
            stream.close()
        } catch (e: IOException) {
            e.printStackTrace()
            on<DefaultAlerts>().thatDidntWork()
            return
        }

        val contentUri = FileProvider.getUriForFile(on<ActivityHandler>().activity!!, MediaHandler.AUTHORITY, file)

        if (contentUri == null) {
            on<DefaultAlerts>().thatDidntWork()
            return
        }

        val shareIntent = Intent()
        shareIntent.action = Intent.ACTION_SEND
        shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
        shareIntent.setDataAndType(contentUri, on<ApplicationHandler>().app!!.contentResolver.getType(contentUri))
        shareIntent.putExtra(Intent.EXTRA_STREAM, contentUri)
        shareIntent.type = "image/jpg"
        on<ActivityHandler>().activity!!.startActivity(Intent.createChooser(shareIntent, on<ResourcesHandler>().resources.getString(R.string.share_with)))
    }
}
